src/System.Text.Primitives/System/Text/InvariantParser_bool.cs

```diff
@@ -122,39 +123,46 @@ public static bool TryParse(byte[] utf8Text, int index, out bool value, out int
                 return false;
             }
 
-            byte firstByte = utf8Text[index];
-
-            if (firstByte == '1')
-            {
-                bytesConsumed = 1;
-                value = true;
-                return true;
-            }
-            else if (firstByte == '0')
-            {
-                bytesConsumed = 1;
-                value = false;
-                return true;
-            }
-            else if (isTrue(utf8Text, index))
-            {
-                bytesConsumed = 4;
-                value = true;
-                return true;
-            }
-            else if (isFalse(utf8Text, index))
-            {
-                bytesConsumed = 5;
-                value = false;
-                return true;
-            }
-            else
+            if (cultureAndEncodingInfo.IsInvariantUtf8)
             {
-                return false;
+
+                byte firstByte = utf8Text[index];
+
+                if (firstByte == '1')
+                {
+                    bytesConsumed = 1;
+                    value = true;
+                    return true;
+                }
+                else if (firstByte == '0')
+                {
+                    bytesConsumed = 1;
+                    value = false;
+                    return true;
+                }
+                else if (isTrue(utf8Text, index))
```

Comment:
I know it's not part of the review, but these method names should be capitalized.